src/NovelRT/Graphics/ImageRect.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#include <NovelRT.h>
+
+namespace NovelRT::Graphics {
+  ImageRect::ImageRect(const Transform& transform,
+    int layer,
+    ShaderProgram shaderProgram,
+    Camera* camera,
+    const std::string& imageDir,
+    const RGBAConfig& colourTint) :
+    RenderObject(transform,
+      layer,
+      shaderProgram,
+      camera),
+    _imageDir(imageDir),
+    _textureId(Utilities::Lazy<GLuint>([] {
+    GLuint tempTexture;
+    glGenTextures(1, &tempTexture);
+    return tempTexture;
+      })),
+    _uvBuffer(Utilities::Lazy<GLuint>(generateStandardBuffer)),
+        _colourTintBuffer(Utilities::Lazy<GLuint>(generateStandardBuffer)),
+        _colourTint(colourTint),
+        _logger(Utilities::Misc::CONSOLE_LOG_GFX) {}
+
+      ImageRect::ImageRect(const Transform& transform,
+        int layer,
+        ShaderProgram shaderProgram,
+        Camera* camera,
+        const RGBAConfig& colourTint) : ImageRect(transform, layer, shaderProgram, camera, "", colourTint) {
+      }
+
+      void ImageRect::drawObject() {
+        if (!getActive())
+          return;
+
+        glUseProgram(_shaderProgram.shaderProgramId);
+        glBindBuffer(GL_UNIFORM_BUFFER, _shaderProgram.finalViewMatrixBufferUboId);
+        glBufferData(GL_UNIFORM_BUFFER, sizeof(Maths::GeoMatrix4<float>), &_finalViewMatrixData.getActual(), GL_STATIC_DRAW);
+
+        glBindTexture(GL_TEXTURE_2D, _textureId.getActual());
+        glBindVertexArray(_vertexArrayObject.getActual());
+        glEnableVertexAttribArray(0);
+        glBindBuffer(GL_ARRAY_BUFFER, _vertexBuffer.getActual());
+        glVertexAttribPointer(
+          0,
+          3,
+          GL_FLOAT,
+          GL_FALSE,
+          0,
+          nullptr
+        );
+
+        glEnableVertexAttribArray(1);
+        glBindBuffer(GL_ARRAY_BUFFER, _uvBuffer.getActual());
+        glVertexAttribPointer(
+          1,
+          2,
+          GL_FLOAT,
+          GL_FALSE,
+          0,
+          nullptr
+        );
+        glEnableVertexAttribArray(2);
+        glBindBuffer(GL_ARRAY_BUFFER, _colourTintBuffer.getActual());
+        glVertexAttribPointer(
+          2,
+          4,
+          GL_FLOAT,
+          GL_FALSE,
+          0,
+          nullptr
+        );
+
+        glDrawArrays(GL_TRIANGLES, 0, 6);
+        glDisableVertexAttribArray(2);
+        glDisableVertexAttribArray(1);
+        glDisableVertexAttribArray(0);
+        glBindVertexArray(0);
+      }
+
+      void ImageRect::configureObjectBuffers() {
+        RenderObject::configureObjectBuffers();
+
+        _uvCoordinates = {
```

Comment:
just a note: a lot of this could be instanced and shared across multiple instances (considering its all "unit sized" and then scaled to the correct size by the shader)

Reply:
Same as before, I think we should look to do these sorts of changes alongside the buffer overhaul.

Reply:
Right, was just calling stuff out as a I reviewed the code (hence `just a note` 😄)